File: Data_Structures/JAVA/src/00Array/Array2.java
public class Array2 {
    
     public static int[] twoNumberSum(int[] array,int target){
        int i = 0;
        int j = array.length-1;
        int temp;
       
           while(i<j){
        while(i<j && array[j]==target){
            j--;
        }
        if(array[i]==target && array[j]!=target){
          temp = array[i];
          array[i] = array[j];
          array[j] = temp; 
          j--;
        }
        i++;
       }
       return array;
}
      
    public static void main(String[] args) {
        int[] arr = {6,1,8,10,4,15,6,3,9,6};
        int[] result = twoNumberSum(arr, 6);
        for(int i=0; i<result.length; i++){
            System.out.println(result[i]);
        }
    }
}
// Here what we have done is :
//we input an array and a target from the user then, we have to print all the 
//targeted number in last positions of the array;

File: Data_Structures/JavaScript/00Arrays/array2.js
//we input an array and a target from the user then, 
//we have to print targeted number in last positions even if it comes
//in multiple time in that array;

const moveToTheLast = (array, target)=>{
 i = 0;
 j = array.length -1;
 while(i < j){
    if(array[i]== target && array[j] == target){
        j--;
    }else if(array[i]== target && array[j]!=target){
        let temp = array[i];
        array[i] = array[j];
        array[j] = temp;
        j--;
    }else if(array[i]!= target && array[j]!= target){
        i++;
    } else{
        i++;
    }
 }
 return array;
}

let array = [5, 4, 5, 14, 3, 6, 5, 5 ]
let result = moveToTheLast(array, 5);
console.log(result);